code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/EdgePersonalizationResponseHandler.java

```diff
@@ -336,16 +467,19 @@ void createInAppMessage(final RuleConsequence triggeredConsequence) {
     private void cacheImageAssetsFromPayload(final JSONObject ruleJsonObject) {
         List<String> remoteAssetsList = new ArrayList<>();
         try {
-            final JSONArray rulesArray = ruleJsonObject.getJSONArray(JSON_KEY);
-            final JSONArray consequence = rulesArray.getJSONObject(0).getJSONArray(JSON_CONSEQUENCES_KEY);
-            final JSONObject details = consequence.getJSONObject(0).getJSONObject(MESSAGE_CONSEQUENCE_DETAIL);
-            final JSONArray remoteAssets = details.getJSONArray(MESSAGE_CONSEQUENCE_DETAIL_KEY_REMOTE_ASSETS);
-            if (remoteAssets.length() != 0) {
-                for (int index = 0; index < remoteAssets.length(); index++) {
-                    final String imageAssetUrl = (String) remoteAssets.get(index);
-                    if (UrlUtils.isValidUrl(imageAssetUrl)) {
-                        Log.debug(LOG_TAG, SELF_TAG, "Image asset to be cached (%s) ", imageAssetUrl);
-                        remoteAssetsList.add(imageAssetUrl);
+            final JSONObject details = MessagingUtils.getConsequenceDetails(ruleJsonObject);
+            final String messageFeedImageUrl = details.optString(MessagingConstants.MessageFeedKeys.IMAGE_URL);
```

Comment:
Let's not do this for now as we just plan to return the feed propositions to the customers. We can look into asset caching and other optimizations once we implement auto-rendering for feeds.